scheduler/pkg/envoy/processor/incremental.go

```diff
@@ -435,6 +435,13 @@ func (p *IncrementalProcessor) addExperiment(exp *experiment.Experiment) error {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 	routeName := fmt.Sprintf("%s.experiment", exp.Name)
+
```

Comment:
Does this code (addExperiment) ever get called if there's no direct change to the experiment itself? I see it's called from `handleExperimentUpdates`, but I can't find model version updates triggering something to publish on that hub topic.

Reply:
Yes check [here](https://github.com/SeldonIO/seldon-core/blob/5383368fc6a62ab9bc27aad28b6617ab41005143/scheduler/pkg/store/experiment/store.go#L173) which will trigger changes to models to update experiments as well. This is also covered by tests.

Reply:
Yep, actually found the place and then just saw your update. Thanks! I did see the test as well, but that is just manually calling `experimentUpdate`

Reply:
oh I see, this was redundant in the test and I remove this explicit call to `experimentUpdate` now.